test/functorch/test_aotdispatch.py

```diff
@@ -2148,24 +2148,29 @@ def setup_context(ctx, inputs, output):
         f.register_fake(f_fake)
         f.register_autograd(backward, setup_context=setup_context)
 
-        def fn(x: torch.Tensor, x1: torch.Tensor) -> torch.Tensor:
-            return torch.ops._test._clone(x, x1)
+        def fn(x: torch.Tensor, x1: torch.Tensor, x2: torch.Tensor) -> torch.Tensor:
+            x2.copy_(x2.mul(5))
+            return torch.ops._test._clone(x, x1) + x2
 
-        inp_x, inp_x1 = torch.randn(3, requires_grad=True), torch.randn(
-            3, requires_grad=False
+        inp_x, inp_x1, inp_x2 = (
+            torch.randn(3, requires_grad=True),
+            torch.randn(3, requires_grad=False),
+            torch.randn(3, requires_grad=False),
         )
 
-        ref_x, ref_x1 = inp_x.clone(), inp_x1.clone()
-        ref_y = f(ref_x, ref_x1)
+        ref_x, ref_x1, ref_x2 = inp_x.clone(), inp_x1.clone(), inp_x2.clone()
+        ref_y = fn(ref_x, ref_x1, ref_x2)
         ref_y.sum().backward()
-        x, x1 = inp_x.clone(), inp_x1.clone()
+        x, x1, x2 = inp_x.clone(), inp_x1.clone(), inp_x2.clone()
         compiled_f = aot_function(fn, nop)
-        y = compiled_f(x, x1)
-        loss = y.sum()
-        loss.backward()
+        y = compiled_f(x, x1, x2)
```

Comment:
it looks like this test mutates `x2` in the forward, and `x1` in the backward.

I think that if you really want to confirm that our mutations are showing up in the right places, you should assert the values of `x`, `x1` and `x2` against their references before **and** after running the backward
```
y = compiled_f(x, x1, x2)

# assert that only x2 was mutated during the fw
self.assertEqual(x, x_ref)
self.assertEqual(x1, x1_ref)
self.assertEqual(x2, x2_ref)
self.assertEqual(y, y_ref)

# now run the backward and assert that x1 is mutated properly
y.sum().backward()

self.assertEqual(x, x_ref)
self.assertEqual(x1, x1_ref)
self.assertEqual(x2, x2_ref)
self.assertEqual(y, y_ref)

# and the gradients are correct
self.assertEqual(x.grad, x_ref.grad)
self.assertEqual(x1.grad, x1_ref.grad)
self.assertEqual(x2.grad, x2_ref.grad)
```